src/FastBoard.h

```diff
@@ -125,6 +125,8 @@ class FastBoard {
     int m_boardsize;
     int m_squaresize;
 
+    std::array<std::array<unsigned short, MAXSQ>, 8> m_symmetry_idx;
```

Comment:
How much of a pain would it be to make it so there is only one of these tables? Or is there a reason why it makes sense to have one for each object?

Reply:
It's tied to the board size, which is variable for this class.

Reply:
Of course. Can totally see it now, thanks.

Reply:
If we can live with an extra indirection you can make it a static class owned vector (indexed by board size) and just check if that is initialized the moment you initialize FastBoard.

Alternatively, do the calculation on the fly...if necessary limit the symmetry check to the first 10-20 moves or so?

That's some of the things I want to look at.